docs/sources/reference/config-blocks/http.md

```diff
@@ -175,3 +190,24 @@ The `client` block is used to check the certificate presented to the server.
 [windows_certificate_filter]: #windows-certificate-filter-block
 [server]: #server-block
 [client]: #client-block
+
+### auth block
+The auth block configures server authentication for the http block. This can be used to enable basic authentication and to set authentication filters for specified API paths.
+
+### basic block
+The basic block enables basic HTTP authentication by requiring both a username and password for access.
+
+| Name                  | Type           | Description                                                       | Default | Required |
+| --------------------- | -------------- | ----------------------------------------------------------------- | ------- | -------- |
+| `username`            | `string`       | The username to use for basic authentication.                     |   N/A   | yes      |
+| `password`            | `string`       | The password to use for basic authentication.                     |   N/A   | yes      |
+
+
+# filter block
+The filter block is used to configure which API paths should be protected by authentication. It allows you to specify a list of paths, using prefix matching, that will require authentication.
+
+| Name                  | Type           | Description                                                                                                            | Default | Required |
+| --------------------- | -------------- | ---------------------------------------------------------------------------------------------------------------------- | ------- | -------- |
+| `path`                | `list(string)` | List of API paths to be protected by authentication. The paths are matched using prefix matching.                      | `[]`    | no       |
+| `auth_if_match`       | `bool`         | If true, authentication is required for all matching paths. If false, authentication is excluded for these paths.      | `true`  | no       |
```

Comment:
IMO this isn't a very clear name, what do you think of `authenticate_matching`?

I also considered `requires_authentication`/`authentication_required`, but they felt a little less direct.

Reply:
Yes we discussed this name a bit but have a hard time to come up with a good name and this was the most explicit one. 

The idea with this flag is to invert the behavior

Require auth on only `/metrics`:
```
filter {
  paht = ["/metrics"]
  auth_if_match = true
}
```

Require auth on _all_ endpoints except `/metrics`:
```
filter {
  paht = ["/metrics"]
  auth_if_match = false
}
```

Reply:
`authenticate_matching` sounds better to me too! WDYT about `authenticate_paths`? I think it might be even better since it's clear that we authenticate the paths above. Or `authenticate_matching_paths`?

Reply:
Yeah I like `authenticate_matching_paths` too.

Reply:
Sounds good to me!